qCC/ccColorFromScalarDlg.cpp

```diff
@@ -44,12 +44,21 @@ ccColorFromScalarDlg::ccColorFromScalarDlg(QWidget* parent, ccPointCloud* pointC
 {
 	m_ui->setupUi(this);
 	
+	
+	if (m_cloud->getCurrentDisplayedScalarFieldIndex() == -1)
+	{
+		m_cloud->setCurrentDisplayedScalarField(0);
+	}
+	m_storedOrigColorScale = m_cloud->getCurrentDisplayedScalarField()->getColorScale();
```

Comment:
I would store 'm_cloud->getCurrentDisplayedScalarField()' in a temporary variable to shorten that block a little.